range_tombstone_list.cc

```diff
@@ -373,15 +373,24 @@ range_tombstone_list::reverter::insert(range_tombstones_type::iterator it, range
     return _dst._tombstones.insert_before(it, new_rt);
 }
 
+// Reserves space for at least one element so that subsequent v.push_back()
+// does not throw if element construction does not throw.
+// Amortizes space expansion so that a series of N calls to amortized_reserve()
+// starting from an empty container takes O(N) time.
+template <typename T>
+void amortized_reserve(T& v) {
+    v.reserve(v.size() + std::max<size_t>(1, v.size()));
+}
```

Comment:
why not log(N)? using growth factor of 1.5-2. although with chunked_vector, new room will only be made once last chunk is exhausted. looks like with this approach growth factor decreases as size increases, with capacity barely increasing the closer the factor is to 1. why not make the growth factor a constant?

Reply:
The growth factor is constant, it's 2x. What do you mean by "why not log(N)"?

Reply:
if size == 1, then capacity becomes 2, ok growth factor of 2.
if size == 2, then capacity becomes 4. ok.
but once size == 3, then capacity becomes 6. growth factor reduced to 1.5
....
if size == 20, then capacity is 40.
if size == 21, then capacity is 42. growth factor at this point is 1.05

if growth factor of *capacity* is 2, then we need log2(N) reallocations.

the problem seems size is being used as reference for calculating new capacity. is that what you really wanted to do?

Reply:
https://godbolt.org/z/9ajc6YsWW

Reply:
Right, will fix.